File: src/test/fizz-buzz-test.c
// fizz-buzz-test.c

#include <stdio.h>
#include <stdbool.h>

bool Div (int x, int y)
{
   return 0 == x%y;
}

int main (void)
{
   int i = 0;
   for ( i = 1 ; i <= 100 ; ++i )
   {
      if (Div(i,3) && !Div(i,5)){printf("Fizz\n");}
      else if (Div(i,5) && !Div(i,3)){printf("Buzz\n");}
      else if (Div(i,3) && Div(i,5)){printf("FizzBuzz\n");}
      else{printf("%d\n",i);}
   }
   return 0;
}

File: scripts/test/fizz-buzz-test.pl
#!/usr/bin/env -S perl -CSDA

# fizz-buzz-test.pl

use v5.32;
use strict;
use warnings;
use experimental 'switch';
use utf8;
use warnings FATAL => "utf8";

1 == scalar(@ARGV) && $ARGV[0] =~ m/^\d+$/ && '0' ne substr($ARGV[0],0,1) && $ARGV[0] >= 1
or die "Error: fizz-buzz-test.pl takes exactly one argument, which must be a positive integer.\n";
printf("%s\n", 'Fizz' x (0 == $ARGV[0] % 3) . 'Buzz' x (0 == $ARGV[0] % 5));
